aws-redis-cluster.yml

```diff
@@ -79,7 +74,7 @@ provision:
     default: ""
   - field_name: node_type
     type: string
-    details: AWS Elasticache node type (see https://aws.amazon.com/elasticache/pricing) Will be computed from cache_size if empty.
+    details: AWS Elasticache node type (see https://aws.amazon.com/elasticache/pricing)
     default: ""
```

Comment:
question: the TF provider documentation says:

> `node_type`: Required unless `global_replication_group_id` is set.

Do we want to set the property as required?

Reply:
We can't set the property as required in `user_inputs` because it would make some legitimate tests fail.
If `node_type` is set to required, old plans relying on `cache_size` will be forced to specify `node_type` in the plan, but [this have strong implications which are explained in the docs](https://github.com/pivotal-cf/docs-csb-aws/pull/161/files#diff-6dbd410f42da6bcdd2fd11f1a2ab3e720a931d2f49c4c468aa6d519a834c435aR35).

Failing to provide a `node_type` value both in the plan and at provision time will result in an error clearly explaining that the value of `node_type` is empty or not valid.

Reply:
I will add a comment to the YAML to ensure that anyone looking at the code will have this context.

Reply:
I ran acceptance_tests against a plan which doesn't include neither `cache_size` or `node_type` it acceptance_tests failed early with the following message:

> Error: "node_type" is required unless "global_replication_group_id" is set. with aws_elasticache_replication_group.redis, on main.tf line 46, in resource "aws_elasticache_replication_group" "redis": 46: resource "aws_elasticache_replication_group" "redis"

We can't make node_type a `required` field because we need to maintain backward compatibility in which customers specify `cache_size` only.
The problem is that we are calculating `node_type` based on `cache_size`'s value in terraform `data.tf`. But cloud-service-broker validates `required` field before executing any terraform code, so it would complain about `node_type` being empty since terraform computation hasn't executed yet.

Reply:
Documented in the manifest next the `node_type` property for future reference: https://github.com/cloudfoundry/csb-brokerpak-aws/pull/774/commits/20eb3176e9869e386756a8b1cd9177e0fa8b5249